helm/templates/cluster-role.yaml

```
@@ -1,13 +1,72 @@
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
-  name: {{ include "kro.fullname" . }}-cluster-role
+  annotations:
+    kubernetes.io/description: |
+{{- if .Values.rbac.legacy }}
+      This ClusterRole grants cluster-wide admin access for the Kro controller.
+{{- else }}
+      This ClusterRole grants access for the Kro controller.
```

tiny nit: we "try" to standardize on using `kro` (lowercase) throughout our documentation and code.
```suggestion
      This ClusterRole grants access for the kro controller.
```

(I just did a quick search and it looks like we still using `Kro` in some other areas of the code base.. we'll have to fix that)